docusaurus/docs/Android/03-ui/06-guides/08-adding-custom-attachments.mdx

```diff
@@ -0,0 +1,169 @@
+# Adding Custom Attachments
+
+## Introduction
+
+By default Stream Chat supports several built-in attachment types like image, video, file and giphy. You can also add your own types of attachments such as location, contact, audio, sticker, etc.
+
+In this guide, we'll demonstrate how to build a date sharing feature. Chat users will be able to pick a date from a calendar, preview their selection in the message input and see the sent attachment within a message in the message list.
+
+This involves doing the following steps:
+1. Customizing the message input in a way that it is capable of sending messages with `date` attachments.
+2. Adding support for `date` attachments in the message list.
+
+:::note
+In this guide, we'll show only the main points concerning custom attachments. Smaller parts will be omitted for the sake of being concise.
+
+You can find the full code from this guide on [GitHub](https://github.com/GetStream/Android-Samples/tree/main/custom-attachments).
+:::
+
+## Sending Date Attachments
+
+To begin with, you'll need to customize [MessageInputView](../04-message-components/04-message-input.mdx) so that the attachment button allows the user to pick dates:
+
+```kotlin
+messageInputView.setAttachmentButtonClickListener { 
```

Comment:
I would add comments here so that it's easier to scan.
Such as:

// Create a date picker
// Add an attachment to the message input when the user selects a date
// Show the date picker

The things being done here are standard Android so I would consider this optional, however sometimes it helps to make the code scannable, otherwise, people can get disinterested.

It's up to you whether you think this is a good suggestion or unnecessary.

Reply:
Sounds good yeah!